js/notesAPI.js

```diff
@@ -0,0 +1,39 @@
+export default class NotesAPI {
+
+    static getAllNotes() { // Retrieves all existing notes
+        
+        // Retrieve all notes associated with key in local storage or create it if it doesn't exist
+        const notes = JSON.parse(localStorage.getItem("notesapp-notes") || "[]")
+
+        return notes.sort((a, b) => { // Return list sorted by updated date
+            return new Date(a.updated) > new Date(b.updated) ? -1 : 1
+        })
+    }
+
+    static saveNote(savedNote) { // Saves updated note + inserts new note
+        const notes = NotesAPI.getAllNotes()
+        const exists = notes.find(note => note.id == savedNote.id) // Does the note-to-save exist?
+
+        if (exists) { // Update existing note
+
+            exists.title = savedNote.title
+            exists.body = savedNote.body
+            exists.updated = new Date().toISOString()
+
+        } else { // Create new note
+
+            savedNote.id = Math.floor(Math.random() * 1000000) // TODO: Create anon function to check if id already exists in notes
+            savedNote.updated = new Date().toISOString()
+            notes.push(savedNote)
+        }
+
+        localStorage.setItem("notesapp-notes", JSON.stringify(notes)) // Save updated notes list to local storage
+    }
```

Comment:
Are you storing the notes to the database anywhere? Maybe I'm missing it?

Reply:
We don't have a database yet, so I'm storing my notes in the browser's local storage for now. Once we have a database set up, I'll make my own collection Schema and store the notes in that